src/main/java/com/kenya/jug/arena/service/UserServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.kenya.jug.arena.service;
+
+import com.kenya.jug.arena.io.UserRequest;
+import com.kenya.jug.arena.io.UserResponse;
+import com.kenya.jug.arena.model.UserEntity;
+import com.kenya.jug.arena.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserResponse createUser(UserRequest request) {
+        if (userRepository.existsByEmail(request.getEmail())) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Email already exists");
```

Comment:
Consider replacing this with a custom exception for example: InvalidUserDetailsException, MalformedPayloadException...etc